modules/core/src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -1236,136 +1237,93 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
-        const hasSigScript = !_.isEmpty(input.script);
-        const hasWitnessScript = !_.isEmpty(input.witness);
-
-        if (!hasSigScript && !hasWitnessScript) {
-          // no sig script or witness data for this input
-          debug('no signature script or witness script data for input %s', idx);
-          return 0;
-        }
-
-        let parsedSigScript;
-        try {
-          parsedSigScript = utxolib.bitgo.parseSignatureScript(transaction.ins[idx]);
-        } catch (e) {
-          return false;
-        }
+      spendAmount += currentAmount;
+      explanation.outputs.push({
+        address: currentAddress,
+        amount: currentAmount,
+      });
+    });
+    explanation.outputAmount = spendAmount;
+    explanation.changeAmount = changeAmount;
 
-        if (hasWitnessScript) {
-          if (!txInfo || !txInfo.unspents) {
-            // segwit txs require input values, cannot validate signatures
-            debug('unable to retrieve input amounts from unspents - cannot validate segwit input signatures');
-            return 0;
-          }
+    // add fee info if available
+    if (params.feeInfo) {
+      explanation.displayOrder.push('fee');
+      explanation.fee = params.feeInfo;
+    }
 
-          // lazily populate unspent values
-          if (_.isEmpty(unspentValues)) {
-            txInfo.unspents.forEach((unspent) => {
-              unspentValues[unspent.id] = unspent.value;
-            });
-          }
-        }
+    if (_.isInteger(transaction.locktime) && transaction.locktime > 0) {
+      explanation.locktime = transaction.locktime;
+      explanation.displayOrder.push('locktime');
+    }
 
-        const nonEmptySignatures = parsedSigScript.signatures?.filter((sig) => !_.isEmpty(sig)) ?? [];
-        const validSignatures = nonEmptySignatures.map((sig, sigIndex) => {
-          if (_.isEmpty(sig)) {
-            return false;
-          }
+    const prevOutputs = params.txInfo?.unspents.map((u) => toOutput(u, this.network));
 
-          const parentTxId = (Buffer.from(input.hash).reverse() as Buffer).toString('hex');
-          const inputId = `${parentTxId}:${input.index}`;
-          const amount = unspentValues[inputId];
+    // get information on tx inputs
+    const inputSignatures = transaction.ins.map((input, idx): number => {
+      if (!txInfo || txInfo.unspents.length !== transaction.ins.length) {
+        return 0;
+      }
 
-          try {
-            return utxolib.bitgo.verifySignature(transaction, idx, amount, { signatureIndex: sigIndex });
-          } catch (e) {
-            return false;
-          }
-        });
+      if (!prevOutputs) {
+        throw new Error(`invalid state`);
+      }
 
-        return validSignatures.reduce((validCount, isValid) => (isValid ? validCount + 1 : validCount), 0);
-      });
+      try {
+        return utxolib.bitgo
+          .getSignatureVerifications(transaction, idx, txInfo.unspents[idx].value, {}, prevOutputs)
```

Comment:
This `getSignatureVerifications` function is what makes this all work for p2tr, right?

Reply:
yep (the prevOut parameter in particular)